File: backend/app/schemas/remote_files.py
"""Pydantic schemas for remote file system APIs."""
from __future__ import annotations

from pydantic import BaseModel, Field


class RemoteEntryResponse(BaseModel):
    """Sanitized remote file system entry."""

    name: str
    path: str
    is_dir: bool
    size: int
    mtime: float
    mode: int | None = None


class RemoteListResponse(BaseModel):
    """Directory listing response for a remote session."""

    session_id: str
    current_path: str
    parent_path: str | None
    items: list[RemoteEntryResponse]
    total: int


class RemoteStatResponse(BaseModel):
    """Remote stat response."""

    entry: RemoteEntryResponse


class RemoteListQuery(BaseModel):
    """Query shape for remote list requests."""

    session_id: str = Field(min_length=1)
    path: str | None = None


class RemoteStatQuery(BaseModel):
    """Query shape for remote stat requests."""

    session_id: str = Field(min_length=1)
    path: str = Field(min_length=1)


class RemoteMkdirRequest(BaseModel):
    """Request body for remote directory creation."""

    session_id: str = Field(min_length=1)
    path: str = Field(min_length=1)


class RemoteRenameRequest(BaseModel):
    """Request body for remote rename or move."""

    session_id: str = Field(min_length=1)
    old_path: str = Field(min_length=1)
    new_path: str = Field(min_length=1)


class RemoteDeleteRequest(BaseModel):
    """Request body for remote delete."""

    session_id: str = Field(min_length=1)
    path: str = Field(min_length=1)
    recursive: bool = True

File: frontend/src/api/remoteFiles.ts
import type { RemoteListResponse } from './types';
import { http } from './http';

export async function listRemoteFiles(sessionId: string, path?: string): Promise<RemoteListResponse> {
  const { data } = await http.get<RemoteListResponse>('/api/remote-files/list', {
    params: {
      session_id: sessionId,
      path,
    },
  });
  return data;
}

export async function createRemoteDirectory(sessionId: string, path: string): Promise<void> {
  await http.post('/api/remote-files/mkdir', {
    session_id: sessionId,
    path,
  });
}

export async function renameRemotePath(sessionId: string, oldPath: string, newPath: string): Promise<void> {
  await http.post('/api/remote-files/rename', {
    session_id: sessionId,
    old_path: oldPath,
    new_path: newPath,
  });
}

export async function deleteRemotePath(sessionId: string, path: string, recursive = true): Promise<void> {
  await http.post('/api/remote-files/delete', {
    session_id: sessionId,
    path,
    recursive,
  });
}
